### DISCUSSION TITLE
Problem with building current release

### DISCUSSION THREAD

**Discussion by SirCumAlot1988** [Komet/MediaElch#1392] [Q&A]
Hello,

I'm trying to build the current release (master branch) and followed all the steps explained under 
https://mediaelch.github.io/mediaelch-doc/contributing/build/windows.html

However, if I try to run it (by pressing strg+r in MediaElch.pro), it gives a system error and says that libquazip5.dll could not be found even though I placed it under C:\Qt\Tools\mingw810_32\lib\gcc\i686-w64-mingw32\8.1.0 as explained in the documentation.

If I build the project by choosing Build ==> Build Project "MediaElch" it gives the following error 32 times:

Der Befehl "sed" ist entweder falsch geschrieben oder konnte nicht gefunden werden.

Which means in English:
The command "sed" has been misspelled or could not be found

Strangely the IDE does not point me to a specific line in the source code for this error.

Any ideas what might be the problem here?

Kind regards

**Comment by bugwelle** [comment]
> Der Befehl "sed" ist entweder falsch geschrieben oder konnte nicht gefunden werden.

You can ignore that warning. I get it as well :-)

> Strangely the IDE does not point me to a specific line in the source code for this error.

That's because it's an internal issue of QtCreator (something around QMake files I think).

> However, if I try to run it (by pressing strg+r in MediaElch.pro), it gives a system error and says that libquazip5.dll could not be found even though I placed it under C:\Qt\Tools\mingw810_32\lib\gcc\i686-w64-mingw32\8.1.0 as explained in the documentation.

Have you initialized the git submodules? And did you pass any arguments to QMake, e.g. `USE_EXTERN_QUAZIP`?
Because if you followed the instructions and initialized the Git submodule, it _should_ work just fine when pressing Ctrl+R.

Regards,
Andre

**Reply by SirCumAlot1988** [reply]
Hi Andre,

Many thanks for your support!

> Have you initialized the git submodules?

Yes, I ran _git submodule update --init_  in the main project folder.

>And did you pass any arguments to QMake, e.g. `USE_EXTERN_QUAZIP`? 

I added _CONFIG += USE_EXTERN_QUAZIP_ to MediaElch.pro as first line.

Furthermore, I changed line 9 and 10 in \src\export\ExportTemplateLoader.cpp to 

_#include "quazip5/quazip.h"
#include "quazip5/quazipfile.h"_

because it gives a file not found error otherwise.

Unfortuantely, it still doesn't work. Keeps saying that libquazip5.dll could not be found.

Kind regards

**Reply by bugwelle** [reply]
hmm... Ok. Could you try to remove `CONFIG += USE_EXTERN_QUAZIP`, i.e. configure MediaElch to use the bundled QuaZip module? It may be that the build documentation is outdated. It was updated by a contributor who is a Windows user. I myself, used the following steps:

Steps 1 to 4 of the build documentation, then open `MediaElch.pro` and build the project.

**Reply by bugwelle** [reply]
@AldanaFan I've updated the build instructions: https://mediaelch.github.io/mediaelch-doc/contributing/build/windows.html

Tested on a fresh Windows 10 installation. Please make sure to install MinGW 8 in 64bit mode. You can start the Qt Maintenance Tool again to check. :-)

**Comment by SirCumAlot1988** [comment]
Still doesn't work. Now it gives a lot of "undefined reference" errors (e.g. "undefined reference to `get_crc_table@0'").

**Reply by bugwelle** [reply]
Ok. I'll rewrite the build documentation and will re-check it on a fresh Windows 10 Education installation. :-)

**Reply by bugwelle** [reply]
Hi, I've updated the build instructions again. Contains a few more screenshots. Again, tested on a clean Windows 10 installation.

If it still doesn't work, could you provide me the exact steps you took as well as the build log (with the errors)?

**Comment by SirCumAlot1988** [comment]
I'm sorry, it still doesn't work. Seems like a header file is missing.

Here is the error message:

> MediaInfoDLL/MediaInfoDLL.h: No such file or directory

I followed all the steps in your new build instructions.

**Reply by bugwelle** [reply]
Hi,

in your `MediaElch` directory, what are the contents of the `MediaInfoDLL` folder? See step [Other LIbraries](https://mediaelch.github.io/mediaelch-doc/contributing/build/windows.html#other-libraries).

It may be that the correct files are missing.
